app/src/main/java/org/schabi/newpipe/player/VideoPlayer.java

```diff
@@ -178,54 +153,37 @@ private static boolean containsCaseInsensitive(final List<String> list, final St
         return false;
     }
 
-    public void setup(final View view) {
-        initViews(view);
-        setup();
+    public void setup(@NonNull final PlayerBinding binding) {
+        initViews(binding);
+        if (simpleExoPlayer == null) {
+            initPlayer(true);
+        }
+        initListeners();
     }
 
-    public void initViews(final View view) {
-        this.rootView = view;
-        this.surfaceView = view.findViewById(R.id.surfaceView);
-        this.surfaceForeground = view.findViewById(R.id.surfaceForeground);
-        this.loadingPanel = view.findViewById(R.id.loading_panel);
-        this.endScreen = view.findViewById(R.id.endScreen);
-        this.controlAnimationView = view.findViewById(R.id.controlAnimationView);
-        this.controlsRoot = view.findViewById(R.id.playbackControlRoot);
-        this.currentDisplaySeek = view.findViewById(R.id.currentDisplaySeek);
-        this.playerTopShadow = view.findViewById(R.id.playerTopShadow);
-        this.playerBottomShadow = view.findViewById(R.id.playerBottomShadow);
-        this.playbackSeekBar = view.findViewById(R.id.playbackSeekBar);
-        this.playbackCurrentTime = view.findViewById(R.id.playbackCurrentTime);
-        this.playbackEndTime = view.findViewById(R.id.playbackEndTime);
-        this.playbackLiveSync = view.findViewById(R.id.playbackLiveSync);
-        this.playbackSpeedTextView = view.findViewById(R.id.playbackSpeed);
-        this.bottomControlsRoot = view.findViewById(R.id.bottomControls);
-        this.topControlsRoot = view.findViewById(R.id.topControls);
-        this.qualityTextView = view.findViewById(R.id.qualityTextView);
-
-        this.subtitleView = view.findViewById(R.id.subtitleView);
+    public void initViews(@NonNull final PlayerBinding binding) {
+        playerBinding = binding;
+        playbackSeekBar = (SeekBar) binding.playbackSeekBar;
+        qualityTextView = (TextView) binding.qualityTextView;
+        playbackSpeed = (TextView) binding.playbackSpeed;
```

Comment:
Why did you keep these three variables?

Reply:
I was getting compilation errors as those three views were being stored as `View` variables for some reason.